app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
> Oh, that's actually what puts the "dynamic" in "dynamic server"--it's running the HTML renderer for each request and sending the generated response to the client without ever storing it in the file system (since that would just slow things down). That's basically what's happening in lines 45-64 of `app.js`, when the server calls `indexStatic.generate()`.

heh, sorry I think I phrased my question wrong - I understand the "dynamic" server being "dynamic" part :stuck_out_tongue: but I was wondering how template files/JS gets compiled from `.jsx`.  I assume React handles the compilation?